### PR TITLE
Added support for color schemes

### PR DESCRIPTION
**Pull request by J9465812**

### CODE REVIEW

File: src/main/java/com/sarahhand/fractals/model/MandelbrotConfig.java [hunk 1]
```diff
@@ -52,6 +56,15 @@ public ColorPalette getPalette() {
 		return palette;
 	}
 	
+	@JsonIgnore
+	public ColorScheme getColorScheme(){
+		return colorScheme;
+	}
+
+	public void setColorScheme(ColorScheme colorScheme){
```
Comment (SarahHand) [inline src/main/java/com/sarahhand/fractals/model/MandelbrotConfig.java:64]: I don't think we should have a set method for this. We don't have set methods for any of the other properties.

File: src/main/java/com/sarahhand/fractals/model/MandelbrotConfig.java [hunk 2]
```diff
@@ -85,7 +99,8 @@ public MandelbrotConfig(Double center, double zoom, int maxDwell, ColorPalette p
 	public MandelbrotConfig(@JsonProperty("x") double x, @JsonProperty("y") double y, 
 			@JsonProperty("zoom") double zoom, 
 			@JsonProperty("maxDwell") int maxDwell, 
-			@JsonProperty("palette") ColorPalette palette) {
+			@JsonProperty("palette") ColorPalette palette,
+			@JsonProperty("colorScheme") ColorScheme colorScheme) {
```
Comment (SarahHand) [inline src/main/java/com/sarahhand/fractals/model/MandelbrotConfig.java:103]: I think we should remove colorScheme from this constructor until we figure out how to get Jackson parsing of the colorScheme working. Can you remove this argument for now?

File: src/main/java/com/sarahhand/fractals/model/MandelbrotPointData.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Represents the data needed for a color scheme to color a point in the Mandelbrot set
+ * @author J9465812
+ */
+public class MandelbrotPointData implements PointData{
+	
+	private Double location;
+	private int escapeTime;
+	private Double z;
```
Comment (SarahHand) [inline src/main/java/com/sarahhand/fractals/model/MandelbrotPointData.java:13]: What is z?

File: src/main/java/com/sarahhand/fractals/model/MandelbrotPointData.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	public MandelbrotPointData(Double location, int escapeTime, Double z){
+		this.location = location;
+		this.escapeTime = escapeTime;
+		this.z = z;
+	}
+	
+	public Double getLocation(){
+		return location;
+	}
+	public void setLocation(Double location){
```
Comment (SarahHand) [inline src/main/java/com/sarahhand/fractals/model/MandelbrotPointData.java:24]: Does this class need to have any set methods?  If there isn't a strong reason to keep them, I'd get rid of them.

File: src/main/java/com/sarahhand/fractals/model/colorscheme/ColorScheme.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+import com.sarahhand.fractals.model.FractalConfig;
+import com.sarahhand.fractals.model.PointData;
+
+public interface ColorScheme{
+	
+	public String getName();
+	
+	public Color getColor(PointData data, FractalConfig config);
+}
```
Comment (M00031) [inline src/main/java/com/sarahhand/fractals/model/colorscheme/ColorScheme.java:13]: Add Javadoc to this interface.

**Review by SarahHand — CHANGES_REQUESTED** [review]

File: src/main/java/com/sarahhand/fractals/model/colorscheme/mandelbrotset/EscapeTimeColorScheme.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		b1 = c1.getBlue();
+		b2 = c2.getBlue();
+		
+		double r = r1*(1 - amount) + r2*amount;
+		double g = g1*(1 - amount) + g2*amount;
+		double b = b1*(1 - amount) + b2*amount;
+		
+		return new Color((int)r, (int)g, (int)b);
+	}
+}
```
Comment (M00031) [inline src/main/java/com/sarahhand/fractals/model/colorscheme/mandelbrotset/EscapeTimeColorScheme.java:76]: Add Javadoc to the class and the interpolate method.

File: src/main/java/com/sarahhand/fractals/viewer/MandelbrotViewerImpl.java [hunk 7]
```diff
@@ -218,44 +227,11 @@ private Color getPointCol(Double p){
...
-		
-		double r = r1*(1 - amount) + r2*amount;
-		double g = g1*(1 - amount) + g2*amount;
-		double b = b1*(1 - amount) + b2*amount;
-		
-		return new Color((int)r, (int)g, (int)b);
+		return config.getColorScheme().getColor(new MandelbrotPointData(p, -1, new Double(z.x, z.y)), config);
 	}
 	
 	private static class ComplexNumber{
```
Comment (M00031) [inline src/main/java/com/sarahhand/fractals/viewer/MandelbrotViewerImpl.java:237]: Add Javadoc to this class and all of the methods in this class.

**Review by M00031 — CHANGES_REQUESTED** [review]

**Review by SarahHand — APPROVED** [review]

**Review by M00031 — APPROVED** [review]

File: src/main/java/com/sarahhand/fractals/viewer/MandelbrotViewerImpl.java (hunk 7 above)
Comment (J9465812) [reply src/main/java/com/sarahhand/fractals/viewer/MandelbrotViewerImpl.java:237]: The class already has Javadoc, and all the method either have self-explanatory names or get Javadoc from the interface. So I don't think any more Javadoc is necessary.

Comment (M00031) [reply src/main/java/com/sarahhand/fractals/viewer/MandelbrotViewerImpl.java:237]: That makes sense.

**Pull request merged**